### ISSUE TITLE
docker: Error response from daemon: OCI runtime create failed

### ISSUE THREAD

**Issue by murthy95** [NVIDIA/nvidia-docker#1245]
I face the following error when i try to run the cuda test image. I am running on ubuntu16.04 and has docker 19.03 installed. I am unable to run  any container using --gpus option.

`docker: Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:449: container init caused \"process_linux.go:432: running prestart hook 0 caused \\\"error running hook: exit status 1, stdout: , stderr: nvidia-container-cli: initialization error: cuda error: no cuda-capable device is detected\\\\n\\\"\"": unknown.`

I have the latest driver installed 440.64.00 and I installed nvidia-container-toolkit using the instructions from [here](https://github.com/NVIDIA/nvidia-docker#nvidia-container-toolkit) on my ubuntu 16.04 machine.

 - [ ] Some nvidia-container information: `nvidia-container-cli -k -d /dev/tty info`
```
I0409 06:56:28.648342 1888 nvc.c:281] initializing library context (version=1.0.7, build=b71f87c04b8eca8a16bf60995506c35c937347d9)
I0409 06:56:28.648523 1888 nvc.c:255] using root /
I0409 06:56:28.648540 1888 nvc.c:256] using ldcache /etc/ld.so.cache
I0409 06:56:28.648558 1888 nvc.c:257] using unprivileged user 65534:65534
W0409 06:56:28.648639 1888 nvc.c:166] skipping kernel modules load due to user namespace
I0409 06:56:28.648934 1889 driver.c:133] starting driver service
I0409 06:56:28.657759 1888 driver.c:233] driver service terminated with signal 15
nvidia-container-cli: initialization error: cuda error: no cuda-capable device is detected
```
 - [ ] Kernel version from `uname -a`
`Linux ------------- 5.3.0-42-generic #34~18.04.1-Ubuntu SMP Fri Feb ---- UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`
 - [ ] Driver information from `nvidia-smi -a`
```
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 440.64.00    Driver Version: 440.64.00    CUDA Version: 10.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
```
 - [ ] Docker version from `docker version`
Docker version 19.03.8, build afacb8b7f0

 - [ ] NVIDIA packages version from `dpkg -l '*nvidia*'` _or_ `rpm -qa '*nvidia*'`
```
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version      Architecture Description
+++-==============-============-============-=================================
ii  libnvidia-cont 1.0.7-1      amd64        NVIDIA container runtime library
ii  libnvidia-cont 1.0.7-1      amd64        NVIDIA container runtime library
ii  nvidia-440     440.64.00-0u amd64        NVIDIA binary driver - version 44
ii  nvidia-440-dev 440.64.00-0u amd64        NVIDIA binary Xorg driver develop
un  nvidia-contain <none>       <none>       (no description available)
un  nvidia-contain <none>       <none>       (no description available)
ii  nvidia-contain 1.0.5-1      amd64        NVIDIA container runtime hook
un  nvidia-current <none>       <none>       (no description available)
un  nvidia-driver- <none>       <none>       (no description available)
un  nvidia-legacy- <none>       <none>       (no description available)
un  nvidia-libopen <none>       <none>       (no description available)
un  nvidia-libopen <none>       <none>       (no description available)
ii  nvidia-modprob 440.64.00-0u amd64        Load the NVIDIA kernel driver and
un  nvidia-opencl- <none>       <none>       (no description available)
ii  nvidia-opencl- 440.64.00-0u amd64        NVIDIA OpenCL ICD
un  nvidia-persist <none>       <none>       (no description available)
ii  nvidia-prime   0.8.2        amd64        Tools to enable NVIDIA's Prime
ii  nvidia-setting 440.64.00-0u amd64        Tool for configuring the NVIDIA g
un  nvidia-setting <none>       <none>       (no description available)
un  nvidia-smi     <none>       <none>       (no description available)
un  nvidia-vdpau-d <none>       <none>       (no description available)
```
 - [ ] NVIDIA container library version from `nvidia-container-cli -V`
```
version: 1.0.7
build date: 2020-01-21T19:00+00:00
build revision: b71f87c04b8eca8a16bf60995506c35c937347d9
build compiler: gcc-5 5.4.0 20160609
build platform: x86_64
build flags: -D_GNU_SOURCE -D_FORTIFY_SOURCE=2 -DNDEBUG -std=gnu11 -O2 -g -fdata-sections -ffunction-sections -fstack-protector -fno-strict-aliasing -fvisibility=hidden -Wall -Wextra -Wcast-align -Wpointer-arith -Wmissing-prototypes -Wnonnull -Wwrite-strings -Wlogical-op -Wformat=2 -Wmissing-format-attribute -Winit-self -Wshadow -Wstrict-prototypes -Wunreachable-code -Wconversion -Wsign-conversion -Wno-unknown-warning-option -Wno-format-extra-args -Wno-gnu-alignof-expression -Wl,-zrelro -Wl,-znow -Wl,-zdefs -Wl,--gc-sections
```
 - [ ] NVIDIA container library logs (see [troubleshooting](https://github.com/NVIDIA/nvidia-docker/wiki/Troubleshooting))
```
-- WARNING, the following logs are for debugging purposes only --

I0408 11:20:16.021933 850 nvc.c:281] initializing library context (version=1.0.7, build=b71f87c04b8eca8a16bf60995506c35c937347d9)
I0408 11:20:16.022240 850 nvc.c:255] using root /
I0408 11:20:16.022310 850 nvc.c:256] using ldcache /etc/ld.so.cache
I0408 11:20:16.022405 850 nvc.c:257] using unprivileged user 65534:65534
W0408 11:20:16.022518 850 nvc.c:166] skipping kernel modules load due to user namespace
I0408 11:20:16.022867 855 driver.c:133] starting driver service
I0408 11:20:16.049609 850 driver.c:233] driver service terminated with signal 15
```
 - [ ] Docker command, image and tag used
`docker run --gpus all nvidia/cuda:10.0-base nvidia-smi`

**Comment by amwfarid** [comment]
I confirm the same issue for Centos7. Looking forward to a solution soon

**Comment by amwfarid** [comment]
A quick update: I resolved my issue after uninstalling/reinstalling nvidia GPU driver (418.87 and Cuda 10.1 in my case).

I was looking into other issues in this repo and came across this tip to run this command as sudo:

`sudo nvidia-container-cli -k -d /dev/tty info`

There I realized that some modules were not loading properly, which was why the cli wasn't able to start. So I resorted to just doing a clean install and all went well for me.

I hope this might be helpful.
